spanner/leaderboard/complete/src/main/java/com/google/codelabs/App.java

```diff
@@ -0,0 +1,353 @@
+    } else if (insertType.equals("scores")) {
+      // Insert scores.
+      insertScores(dbClient);
+    } else {
+      // Invalid input.
+      System.out.println("Invalid value for 'type of insert'. "
+          + "Specify a valid value: 'players' or 'scores'.");
+      System.exit(1);
+    }
+  }
+
+  static void insertPlayers(DatabaseClient dbClient) {
+    dbClient
+        .readWriteTransaction()
+        .run(
+            new TransactionCallable<Void>() {
+              @Override
+              public Void run(TransactionContext transaction) throws Exception {
+                // Get the number of players.
+                String sql =
+                    "SELECT Count(PlayerId) as PlayerCount FROM Players";
+                ResultSet resultSet = transaction.executeQuery(Statement.of(sql));
+                long numberOfPlayers = 0;
+                while (resultSet.next()) {
+                  numberOfPlayers = resultSet.getLong("PlayerCount");
+                }
+                // Insert 100 player records into the Players table.
+                long randomId;
+                for (int x = 1; x <= 100; x++) {
+                  numberOfPlayers++;
+                  randomId = (long) Math.floor(Math.random() * 9_000_000_000L) + 1_000_000_000L;
+                  transaction.buffer(
+                      Mutation.newInsertBuilder("Players")
+                          .set("PlayerId")
+                          .to(randomId)
+                          .set("PlayerName")
+                          .to("Player " + numberOfPlayers)
+                          .build());
+                }
+                return null;
+              }
+            });
+    System.out.println("Done inserting player records...");
+  }
+
+  static void insertScores(DatabaseClient dbClient) {
+    boolean playerRecordsFound = false;
+    ResultSet resultSet =
+        dbClient
+            .singleUse()
+            .executeQuery(Statement.of("SELECT * FROM Players"));
+    while (resultSet.next()) {
+      if (!playerRecordsFound) {
+        playerRecordsFound = true;
+      }
+      final long playerId = resultSet.getLong("PlayerId");
+      dbClient
+          .readWriteTransaction()
+          .run(
+              new TransactionCallable<Void>() {
+                @Override
+                public Void run(TransactionContext transaction) throws Exception {
+                  // Initialize objects for random Score and random Timestamp
+                  LocalDate endDate = LocalDate.now();
+                  long end = endDate.toEpochDay();
+                  int startYear = endDate.getYear() - 2;
+                  int startMonth = endDate.getMonthValue();
+                  int startDay = endDate.getDayOfMonth();
+                  LocalDate startDate = LocalDate.of(startYear, startMonth, startDay);
+                  long start = startDate.toEpochDay();
+                  long randomDay;
+                  LocalDate randomDayDate;
```

Comment:
Do these variables need to be scoped up here? They don't seem to be used anywhere else besides in the function below.

Reply:
Updated.